### PR TITLE
fix: get old season badges ipfs

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**
Based on this ticket https://discord.com/channels/896685581487210577/1047444573397450832/1047445262269952080
Tp trader/lp Badges (based on eth price on Thursday) submitted to the contract were different than what the subgraph was getting based on the current eth price.

This takes the badges shown for already-finished seasons from IPFS (i.e. what was submitted)

### CODE REVIEW

File: src/utils/communitySbt/getSubgraphBadges.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                        awardedTimestampMs: toMillis(parseInt(badge.awardedTimestamp)),
+                        mintedTimestampMs: toMillis(parseInt(badge.mintedTimestamp)),
+                    });
+                }
+            }
+        }
+        
+        return badgesResponse;
+    } catch (error) {
+      return [];
```
Comment (ssbarbee) [inline src/utils/communitySbt/getSubgraphBadges.ts:68]: dont forget to track on Sentry when an error occurs

File: src/entities/communitySbt.ts [hunk 2]
```diff
@@ -315,52 +316,104 @@ class SBT {
...
+        });
+        const mapBadges = new Map<string, BadgeResponse>();
+        badgesResponse.forEach((entry) => {
+            mapBadges.set(entry.id, entry);
+        })
+
+        const network = (await this.provider.getNetwork()).name;
+        const data = await axios.get(geLeavesIpfsUri(network, seasonId));
+
+        const snaphots : Array<{
```
Comment (arturbeg) [inline src/entities/communitySbt.ts:373]: small typo "snapshots"

File: src/entities/communitySbt.ts [hunk 3]
```diff
@@ -315,52 +316,104 @@ class SBT {
...
+            userId,
+            seasonId,
+            seasonStart,
+            seasonEnd,
+        });
+        return badges;
+
+    } 
+    
+    public async getOldSeasonBadges({
```
Comment (arturbeg) [inline src/entities/communitySbt.ts:341]: is there a unit test for this function?

**Comment by arturbeg** [comment]
Worth QAing this with the user who submitted the original support ticket, if you had infinite time what test case scenarios would you write alongside this PR @ioanaroceanu2000 ?

**Comment by arturbeg** [comment]
btw, should this also be replicated in sdk v2 once merged?

File: src/utils/communitySbt/getSubgraphBadges.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import { BadgeResponse } from "../../entities";
+import { ApolloClient, gql, HttpLink, InMemoryCache } from '@apollo/client';
+import { SubgraphBadgeResponse } from "../../entities/communitySbt";
+import { toMillis } from "./helpers";
+
+export async function getSubgraphBadges({
```
Comment (arturbeg) [inline src/utils/communitySbt/getSubgraphBadges.ts:6]: feels like this needs testing?

**Comment by ioanaroceanu2000** [comment]
> btw, should this also be replicated in sdk v2 once merged?
@arturbeg 
I think a lot of things have changed to this SDK (mainly community stuff). Not sure of Costin has updated sdk v2 along the way but probably not.

File: src/utils/communitySbt/getSubgraphBadges.ts (hunk 4 above)
Comment (ioanaroceanu2000) [reply src/utils/communitySbt/getSubgraphBadges.ts:6]: yes, it does. but for new, as a reassurance, this is a copy of the deleted bits from `entities/communitySbt/getSeasonBadges()`

File: src/utils/communitySbt/getSubgraphBadges.ts (hunk 1 above)
Comment (ioanaroceanu2000) [reply src/utils/communitySbt/getSubgraphBadges.ts]: thanks! added in [5213743](https://github.com/Voltz-Protocol/v1-sdk/pull/119/commits/5213743e04b4c717a28d20e4dc7a05472a0da321)

File: src/entities/communitySbt.ts (hunk 3 above)
Comment (ioanaroceanu2000) [reply src/entities/communitySbt.ts:346]: no tests, we'll have to QA this internally

File: src/utils/communitySbt/seasonsConfig.ts [hunk 5]
```diff
@@ -7,4 +7,9 @@ export const goerliSeasonLeavesCid: Record<number, string> = {
 export const mainnetSeasonLeavesCid:  Record<number, string>  = {
     0: "QmefHAb6KEY8jjYCQNz4TPFLLRfD9SadfZpBpVSPHJUQGm",
     1: ""
+}
+
+export const mainnetSeasonBadgesCid:  Record<number, string>  = {
+    0: "QmefHAb6KEY8jjYCQNz4TPFLLRfD9SadfZpBpVSPHJUQGm",
```
Comment (ioanaroceanu2000) [inline src/utils/communitySbt/seasonsConfig.ts:13]: need to modify this once we deploy the leaves

File: src/utils/communitySbt/seasonsConfig.ts [hunk 6]
```diff
@@ -7,4 +7,9 @@ export const goerliSeasonLeavesCid: Record<number, string> = {
 export const mainnetSeasonLeavesCid:  Record<number, string>  = {
     0: "QmefHAb6KEY8jjYCQNz4TPFLLRfD9SadfZpBpVSPHJUQGm",
```
Comment (ioanaroceanu2000) [inline src/utils/communitySbt/seasonsConfig.ts:8]: need to modify this once we deploy the leaves

**Review by arturbeg — APPROVED** [review]

**Pull request merged**